### PR TITLE
TIMX 339 - scaffold project

### PR DESCRIPTION
**Pull request by ghukill**
### Purpose and background context

Scaffolds the project with:
- python version `3.12`
- renames internal CLI app to `abdiff` (while repository remains `transmogrifier-ab-diff`)
- removed Sentry integration (project is fully local at this time)
- provide an initial sketch of file structures for local output
  - [view rendered form in README](https://github.com/MITLibraries/transmogrifier-ab-diff/tree/TIMX-339-scaffold-project?tab=readme-ov-file#concepts)
- [stub files for future code](https://github.com/MITLibraries/transmogrifier-ab-diff/pull/1/commits/5e22b86d338f7d2acec6781df90679d3d6020991)

### How can a reviewer manually see the effects of these changes?

Nothing much to see yet!  Can run `make test` and `make lint` to see all is passing.

### Includes new or updated dependencies?
YES: ran `make update`

### Changes expectations for external applications?
NO

### What are the relevant tickets?
- https://mitlibraries.atlassian.net/browse/TIMX-339

### Developer
- [X] All new ENV is documented in README
- [X] All new ENV has been added to staging and production environments
- [X] All related Jira tickets are linked in commit message(s)
- [X] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer(s)
- [ ] The commit message is clear and follows our guidelines (not just this PR message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The provided documentation is sufficient for understanding any new functionality introduced
- [ ] Any manual tests have been performed **or** provided examples verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,49 @@ Compare transformed TIMDEX records from two versions (A,B) of Transmogrifier.
 
 # abdiff
 
-TODO...
+`abdiff` is the name of the CLI application in this repository that performs an A/B test of Transmogrifier by
```
Comment (ehanson8) [inline README.md:7]: This is a good high-level overview, but I would leave as-is until we're done given the high likelihood of changes during the development process

File: README.md [hunk 2]
```diff
@@ -1,28 +1,52 @@
...
+│           │   └── transformed
+│           │       ├── a
+│           │       │   ├── alma-2024-01-01-transformed-to-index.json
+│           │       │   └── dspace-2024-03-15-transformed-to-index.json
+│           │       └── b
+│           │           ├── alma-2024-01-01-transformed-to-index.json
+│           │           └── dspace-2024-03-15-transformed-to-index.json
+│           └── 2024-08-23_13-30-00
+                └── # and similar structure here for this run...
+```
```
Comment (jonavellecuerdo) [inline README.md:45]: Hmm, how do you feel about rewriting this section to:


## Concepts

1. **Job**: A job in `abdiff` represents the A/B test for comparing the results from two versions of Transmogrifier. When a job is first created, a JSON file with an initial set of configurations is created in the root of the job folder (`job.json`). These configurations are described below:

   ```shell
   {
     "job_name"   <------- command `job-init` writes
     "version_a_sha"   <------ command 'job-init' writes this
     "version_b_sha"   <------ command 'job-init' writes this
   }
   ```

2. **Run**: A run is the _execution_ of a job. The outputs from a run are fully encapsulated in a nested sub-directory of the job folder, with each run uniquely identified by the timestamp of execution (formatted as X). When a run is executed, the job JSON file is copied into the run folder. Over the course of the run, the copied JSON file (`run.json`) is updated with X.

    This results in a file structure similar to the following:

    ```text
    ├── output
    │   └── test-refactor-1
    │       ├── job.json
    │       └── runs
    │           ├── 2024-08-23_12-10-00
    │           │   ├── collate_ab.parquet
    │           │   ├── diff_ab.parquet
    │           │   ├── metrics_ab.json
    │           │   ├── run.json
    │           │   └── transformed
    │           │       ├── a
    │           │       │   ├── alma-2024-01-01-transformed-to-index.json
    │           │       │   └── dspace-2024-03-15-transformed-to-index.json
    │           │       └── b
    │           │           ├── alma-2024-01-01-transformed-to-index.json
    │           │           └── dspace-2024-03-15-transformed-to-index.json
    │           └── 2024-08-23_13-30-00
                   └── # and similar structure here for this run...
    ```

---

**Note:** For instance, I could see us eventually adding a third item like "Diff" to describe what changes the CLI actually captures. Could also see some of this moving to its own doc under docs eventually but keeping things in the README is good for now. Just thought it'd be nice to move some of what we've decided/discussed (from Jira tickets) into actual docs!

**Review by ehanson8 — APPROVED** [review]
Structure looks good, a few suggestions but approved!

**Review by jonavellecuerdo — CHANGES_REQUESTED** [review]
Scaffold is looking great to me! 😄 I proposed an update to the README -- curious to get your thoughts.

File: README.md (hunk 1 above)
Comment (ghukill) [reply README.md:7]: Good call.  That was my initial intention, but got away from me.  Removing steps 1-7 in favor of keeping it high level for now.

File: README.md (hunk 2 above)
Comment (ghukill) [reply README.md]: I like it!  Thanks for suggesting.  I've taken this, and reworked slightly in [this commit](https://github.com/MITLibraries/transmogrifier-ab-diff/pull/1/commits/0d4a141c146c69b007dcd879f719d17f1b397732).

Comment (jonavellecuerdo) [reply README.md]: Awesome! :D

**Review by jonavellecuerdo — APPROVED** [review]

**Pull request merged**